packages/node_modules/@node-red/editor-client/src/js/ui/actions.js

```diff
@@ -1,47 +1,78 @@
 RED.actions = (function() {
     var actions = {
 
-    }
+    };
 
-    function addAction(name,handler) {
+    function addAction(name,handler,options) {
         if (typeof handler !== 'function') {
             throw new Error("Action handler not a function");
         }
         if (actions[name]) {
             throw new Error("Cannot override existing action");
         }
-        actions[name] = handler;
+        actions[name] = {
+            handler: handler,
+            options: options,
+        };
     }
     function removeAction(name) {
-        delete actions[name];
+        delete actions[name].handler;
     }
     function getAction(name) {
-        return actions[name];
+        return actions[name].handler;
     }
     function invokeAction() {
         var args = Array.prototype.slice.call(arguments);
         var name = args.shift();
         if (actions.hasOwnProperty(name)) {
-            actions[name].apply(null, args);
+            var handler = actions[name].handler;
+            handler.apply(null, args);
         }
     }
     function listActions() {
         var result = [];
+        var missing = [];
         Object.keys(actions).forEach(function(action) {
+            var def = actions[action];
             var shortcut = RED.keyboard.getShortcut(action);
             var isUser = false;
             if (shortcut) {
                 isUser = shortcut.user;
             } else {
                 isUser = !!RED.keyboard.getUserShortcut(action);
             }
+            if (!def.label) {
```

Comment:
Whilst this will work, I had suggested this logic go in `addHandler` so the lookup is done at the point in time the action is registered. But we can leave this here for now.

Reply:
I have tried it.  But this code could not be simply moved to addHandler because actions may be added before initialization of `RED._`.

Reply:
Makes sense - thanks for looking at it. We can stick with what you've got here.